source/Core/Material.cpp

```diff
@@ -23,20 +23,33 @@ namespace   Engine
 
     Material::Material(const std::shared_ptr<ShaderProgram> &shaderprogram) :
         m_name("Default"),
-        m_shaderprogram(shaderprogram)
+        m_shaderprogram(shaderprogram),
+	_texture(nullptr)
     {
         // nothing to do
     }
 
     Material::Material(const aiMaterial *assimpMaterial,
-            const std::shared_ptr<ShaderProgram> &shaderprogram) :
+            const std::shared_ptr<ShaderProgram> &shaderprogram,
+	    const TextureLibrary &texLib) :
         m_name("Default"),
-        m_shaderprogram(shaderprogram)
+        m_shaderprogram(shaderprogram),
+	_texture(nullptr)
     {
         aiString assimpName;
 
         assimpMaterial->Get(AI_MATKEY_NAME, assimpName);
         this->m_name = assimpName.C_Str();
+	unsigned int count = assimpMaterial->GetTextureCount(aiTextureType_DIFFUSE);
```

Comment:
Add issue in order to enable support for other texture type.like the normals one.